### PR TITLE
Continue pruning abseil usage: switch from absl::InlinedVector to std::vector.

### PR DESCRIPTION
**Pull request by ScottTodd**
Also switched to using `IREE_ATTRIBUTE_NOINLINE` from `ABSL_ATTRIBUTE_NOINLINE`.

Generated via a pile of regex (code, includes, build deps) and manual cleanup until builds passed again.

* native modules needed some extra changes to work around calling `iree_hal_buffer_view_shape` with rank 0

### CODE REVIEW

**Comment by benvanik** [comment]
seems fine to me - none of these places are load-bearing in that if you are formatting a tensor to a string then you shouldn't be scared of a single extra heap alloc ;)
there's a few places like in the python bindings that could benefit (for shapes and such), but that's also in the non-load-bearing category and we know we want to rework all of that anyway (it's not a rock-solid production-optimized path)

**Comment by ScottTodd** [comment]
> seems fine to me - none of these places are load-bearing in that if you are formatting a tensor to a string then you shouldn't be scared of a single extra heap alloc ;)
> there's a few places like in the python bindings that could benefit (for shapes and such), but that's also in the non-load-bearing category and we know we want to rework all of that anyway (it's not a rock-solid production-optimized path)

We could make `bindings/` special, and I could just exclude that directory from my regex replacements. At a certain point, being in the same repository means being limited by stuff like this though: https://github.com/google/iree/blob/402550bcbdd47ef96a599c3fdb5bbd071bbfb1a1/build_tools/cmake/iree_copts.cmake#L309-L312

**Comment by benvanik** [comment]
> We could make `bindings/` special, and I could just exclude that directory from my regex replacements.

good point - we can leave the python bindings as-is for now - and then eventually move that linkopts just to the python bindings and out of the defaults

File: iree/modules/check/native_module.cc [hunk 1]
```diff
@@ -216,12 +216,12 @@ class CheckModuleState final {
     auto* lhs = lhs_ref.get();
     auto* rhs = rhs_ref.get();
     size_t lhs_rank = iree_hal_buffer_view_shape_rank(lhs);
-    absl::InlinedVector<int32_t, 6> lhs_shape(lhs_rank);
+    std::vector<int32_t> lhs_shape(lhs_rank);
     IREE_RETURN_IF_ERROR(
         iree_hal_buffer_view_shape(lhs, lhs_rank, lhs_shape.data(), nullptr));
```
Comment (ScottTodd) [inline iree/modules/check/native_module.cc:221]: Ahhhh something weird is happening here. (see failing test [logs](https://source.cloud.google.com/results/invocations/1f9fc746-225e-42cc-918e-52819ed99c67/targets/iree%2Fgcp_ubuntu%2Fbazel%2Flinux%2Fx86-swiftshader%2Fcore%2Fpresubmit/log))

https://en.cppreference.com/w/cpp/container/vector/data
> If size() is 0, data() may or may not return a null pointer.

`lhs_rank` is actually `0`, so the vector size/capacity is `0` and `lhs_shape.data(0` returns `nullptr`

that's getting passed in here: https://github.com/google/iree/blob/1386d2cd284bf8babafdf90a0823afabd0e0c9f8/iree/hal/buffer_view.c#L161-L183

where `IREE_ASSERT_ARGUMENT(out_shape);` complains

**Comment by ScottTodd** [comment]
Alright, I think the builds are on track to go green once Kokoro stops tripping over itself. The latest commit has some real changes, while the others are mostly simple replacements.

**Review by benvanik — APPROVED** [review]

**Pull request merged**